extensions/authy/src/component/OtpList.tsx

```diff
@@ -64,21 +115,46 @@ export function OtpList(props: { isLogin: boolean | undefined; setLogin: (login:
       const services: Otp[] = servicesResponse.authenticator_tokens.map((i) => {
         const seed = decryptSeed(i.encrypted_seed, i.salt, authyPassword);
         return {
-          name: i.original_name ? i.original_name : i.name,
+          name: preferCustomName ? i.name || i.original_name : i.original_name || i.name,
+          accountType: i.account_type,
+          issuer: i.issuer,
+          logo: i.logo,
           digits: i.digits,
           seed: seed,
-          otp: generateTOTP(seed, { digits: i.digits, period: 30 }),
+          generate: () => generateTOTP(seed, { digits: i.digits, period: 30 }),
+          period: 30,
         };
       });
       const apps: Otp[] = appsResponse.apps.map((i) => {
         return {
           name: i.name,
           digits: i.digits,
           seed: i.secret_seed,
-          otp: generateTOTP(encode(Buffer.from(i.secret_seed, "hex")), { digits: i.digits, period: 10 }),
+          generate: () => generateTOTP(encode(Buffer.from(i.secret_seed, "hex")), { digits: i.digits, period: 10 }),
+          period: 10,
         };
       });
-      setState([...services, ...apps]);
+      const updateOtp = () =>
+        setState({
+          timeLeft10: calculateTimeLeft(10),
+          timeLeft30: calculateTimeLeft(30),
+          state: [...services, ...apps].map((item) => ({
+            ...item,
+            otp: item.generate(),
+          })),
+        });
+      setInterval(
```

Comment:
Also we can add sections for apps and 3rd party services

Reply:
It was indeed a cool feature to have, maybe @mathieudutour has some feedback on that topic

Reply:
I think it's pretty cool, and rendering 4 times per second is fine, performance shouldn't be an issue. But I'm wondering why 4 times per second instead of just once per second? Seems like it can't update more than once per second, right?

Reply:
I was trying to get closer to the start of the second, I've tweaked things to only re-render once per second now